src/core/lv_obj.c

```diff
@@ -268,6 +270,8 @@ void lv_obj_add_flag(lv_obj_t * obj, lv_obj_flag_t f)
 void lv_obj_remove_flag(lv_obj_t * obj, lv_obj_flag_t f)
 {
     LV_ASSERT_OBJ(obj, MY_CLASS);
+    if(!(obj->flags & f))
```

Comment:
`!lv_obj_has_flag(obj, f)`

Reply:
looking into it..

Reply:
This will for sure work for single, un-OR-ed flags.

![image](https://github.com/user-attachments/assets/2f29c784-72dd-47f0-9ae3-511d6c4dcebd)

For OR-ed flags that would suggest clearing 2 state bits:
lv_obj_has_flag will return TRUE (causing modification and not return from function) only if both state bits are set. 

This means using this function is logically wrong in this case, since we would like to modify the state if ANY of the flags are set. 

Using lv_obj_has_flag_any:

if(!(lv_obj_has_flag_any(obj, f)))